Develop3

leanbre opened the pull request:
Добавлены автотесты на UI с allure-отчетом

Покрыты вход (со всех источников), регистрация, переходы по разделу конструктора

leanbre committed:
feat: автотесты по UI stellar burgers

leanbre committed:
feat: + allure-отчет
fix: убраны debug-строки для запуска локально

okryukova commented on pom.xml:
⚠️Можно улучшить. Хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройку (если необходима) и как запускать (mvn clean test - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблицы используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

okryukova commented on src/main/java/page/ForgotPasswordPage.java:
```diff
@@ -0,0 +1,34 @@
+
+    // Действия на странице
+    // Открытие страницы восстановления пароля
+    public void openPage() {
```
⛔️Нужно исправить. Для всех методов, описывающих шаги теста, необходимо использовать аннотацию @Step

okryukova commented on src/main/java/model/User.java:
```diff
@@ -0,0 +1,26 @@
+package model;
+
+public class User {
```
⚠️Можно улучшить. Можно использовать @DaTa и @AllArgsConstructor из библиотеки Lombok

okryukova commented on src/test/java/ConstructorTest.java:
```diff
@@ -0,0 +1,73 @@
+    public void before() {
+        RestAssured.baseURI = BASE_URL;
+        // Запуск ChromeDriver
+        webDriver = new ChromeDriver();
```
⛔️Нужно исправить. Тестирование проводиться в двух браузерах. Есть следующие способы выбора браузера:  файл.properties, переменная окружения, системная переменная, фабрика

okryukova commented on src/test/java/RegisterTest.java:
```diff
@@ -0,0 +1,108 @@
+                    .and()
+                    .statusCode(SC_ACCEPTED);
+        }
+    }
```
⛔️Нужно исправить. Браузер нужно закрыть после каждого теста

leanbre committed:
fix: методы в объектах страниц переписаны на аннотацию Step
fix: добавлено закрытие браузера после каждого теста
feat: + перегенерированный allure-отчет

okryukova commented on src/test/java/ConstructorTest.java:
```diff
@@ -0,0 +1,102 @@
+
+    @Parameterized.Parameters(name = "Браузер: {0}")
+    public static Object[] data() {
+        return new Object[] {
```
⛔️Нужно исправить. Не нужно указывать браузер в ТД.  Есть следующие способы выбора браузера:  файл.properties, переменная окружения, системная переменная, фабрика. Поправь, пожалуйста, во всем коде

leanbre replied:
Я вынесла наименование браузера в application.properties, а создание самого драйвера - в фабрику DriverFactory, где, в зависимости от значения переменной, будет создаваться драйвер. Это то, что нужно?

leanbre replied:
@okryukova и отчет allure, он должен быть на какой из переменных? Или это не так важно?

leanbre committed:
fix: Создание драйвера вынесено в DriverFactory
fix: убрана параметризация тестов для браузера
feat: тип браузера вынесен в переменные окружения
feat: добавлено отображение пемеренной окружения в allure-отчете
